tests/v1/engine/test_engine_core_client.py

```diff
@@ -87,6 +93,38 @@ async def loop_until_done_async(client: EngineCoreClient, outputs: dict):
             break
 
 
+async def loop_until_fully_done_async(client: EngineCoreClient,
+                                      outputs: dict,
+                                      timeout: float = 20.0):
+    start_time = time.monotonic()
+
+    while True:
+        # Check for timeout
+        elapsed_time = time.monotonic() - start_time
+        if elapsed_time > timeout:
+            raise TimeoutError(
+                f"loop_until_fully_done_async timed out after {timeout} seconds"
+            )
+
+        engine_core_outputs = (await client.get_output_async()).outputs
+
+        if len(engine_core_outputs) == 0:
+            continue
+
+        # Add outputs to the dict
+        for out in engine_core_outputs:
+            outputs[out.request_id].append(out)
+
+        # Check if all request IDs in outputs have finished
+        for req_id in outputs:
+            request_finished = any(out.finished for out in outputs[req_id])
+            if not request_finished:
+                await asyncio.sleep(0.1)
+                continue
+
+        break
```

Comment:
The `continue` will continue the inner loop which I don't think is intended
```suggestion
        if all(outs and outs[-1].finished for outs in outputs.values()):
            break

        await asyncio.sleep(0.1)
```

Reply:
nice catch! I completely missed that